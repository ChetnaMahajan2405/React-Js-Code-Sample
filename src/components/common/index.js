
export { default as Button } from "./Button";
export { default as CheckBox } from "./CheckBox";
export { default as Header } from "./Header";
export { default as Image } from "./Image";
export { default as TextInput } from "./TextInput";
export { default as Tile } from "./Tile";