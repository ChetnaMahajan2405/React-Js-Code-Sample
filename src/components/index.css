
.form-box {
    margin: auto;
    background-color: #ffffff;
    border: 3px solid #f6f6f6;
    padding-top: 16px; 
    padding-bottom: 25px;
};

@media all and (min-width : 1200px) {
    .form-box {
        width: 400px;
        padding-top: 31px; 
        padding-bottom: 40px;
    }
}

@media all and (min-width : 768px) {
    .form-box {
        width: 400px;
        padding-top: 31px; 
        padding-bottom: 40px;
    }
}

@media all and (min-width: 600px) {
    .form-box {
        width: 400px;
        padding-top: 31px; 
        padding-bottom: 40px;
    }

}  

div .form-box .form-header {
    font-size: 24px;
    color: #414141;
    font-family: "Open Sans";
    font-weight: 600;
    text-align: center;
}

div .form-box .form-subHeading {
    font-size: 14px;
    color: #333333;
    font-family: "Open Sans";
    font-weight: 400;
    text-align: center;
}

div .form-box .input-container { 
    padding-left: 20px;
    padding-right: 20px;
    margin-top: 35px;
}

@media all and (max-width: 600px) {
    div .form-box .input-container  {
        padding-left: 10px;
        padding-right: 10px;
    }

}

.carousel {
    margin-top: 141px;
    display: flex;
    justify-content: space-between;
    flex-wrap: wrap;
}

@media all and (max-width: 600px) {
    .carousel {
        margin-top: 100px;
    }
}


/* Create two equal columns that floats next to each other */
.column {
    float: left;
    width: 50%;
}