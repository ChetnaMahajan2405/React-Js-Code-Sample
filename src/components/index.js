export { default as Forgot } from "./Forgot";
export { default as Home } from "./Home";
export { default as Login } from "./Login";
