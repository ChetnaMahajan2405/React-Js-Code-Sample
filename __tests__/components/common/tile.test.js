import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

// Component to be tested
import { Tile, Image } from '../../../src/components/common';

describe('Tile render()', () => {
    it('Snapshot for Tile', () => {
        const output = shallow(<Tile/>);
        expect(toJson(output)).toMatchSnapshot();
    });

    it('Check for inner element Image in Tile', () => {
        const wrapper = shallow(<Tile/>);
        expect(wrapper.find(Image).exists()).toBeTruthy();
    });

    it('displays the Tile text passed on props', function () {
        const wrapper = shallow(<Tile header="test" text="text"/>);

        expect(wrapper.find('.header').text()).toContain('test');
        expect(wrapper.find('.sub-header').text()).toContain('text');

    }); 

})