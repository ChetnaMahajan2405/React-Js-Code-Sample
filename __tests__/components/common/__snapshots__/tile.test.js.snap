// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile render() Snapshot for Tile 1`] = `
<div
  className="tile"
>
  <Image
    alt="tile"
    className="image"
    onErrorImage=""
    src="assets/images/tile1.jpg"
  />
  <div
    className="header"
  >
     
     
  </div>
  <div
    className="sub-header"
  >
     
     
  </div>
</div>
`;
